src/other/contango/ContangoWand.sol

```diff
@@ -0,0 +1,317 @@
+    }
+
+    /// ----------------- Oracle Governance -----------------
+
+    /// @notice Set a pool as a source in the YieldSpace oracle, as long as:
+    /// - It is a pool known to the Yield Ladle
+    /// - The baseId matches the pool's baseId
+    /// - The quoteId matches the pool's seriesId
+    function setYieldSpaceOracleSource(bytes6 seriesId) external auth {
+        IPool pool_ = IPool(yieldLadle.pools(seriesId));
+        require(address(pool_) != address(0), "Pool not known to the Yield Ladle");
+        DataTypes.Series memory series_ = yieldCauldron.series(seriesId);
+        require(address(series_.fyToken) != address(0), "Series not known to the Yield Cauldron");
+        require(address(series_.fyToken) == address(pool_.fyToken()), "fyToken mismatch"); // Sanity check
+
+        yieldSpaceOracle.setSource(seriesId, series_.baseId, pool_);
+    }
+
+    /// @notice Set the YieldSpace oracle as the source for a given asset pair in the Composite oracle, provided the source is set in the YieldSpace oracle
+    function setCompositeOracleSource(bytes6 baseId, bytes6 ilkId) external auth {
+        DataTypes.Series memory series_ = yieldCauldron.series(ilkId);
+        if (series_.fyToken != IFYToken(address(0))) {
+            (IPool pool_,) = yieldSpaceOracle.sources(baseId, ilkId);
+            require(address(pool_) != address(0), "YieldSpace oracle not set");
+            compositeOracle.setSource(baseId, ilkId, yieldSpaceOracle);
+        } else if (yieldCauldron.assets(ilkId) != address(0)) {
+            DataTypes.SpotOracle memory spotOracle_ = yieldCauldron.lookupSpotOracle(baseId, ilkId);
+            compositeOracle.setSource(baseId, ilkId, spotOracle_.oracle);
+        }
+    }
+
+    /// @notice Set a path in the Composite oracle, as long as the path is not overwriting anything
+    function setCompositeOraclePath(bytes6 baseId, bytes6 quoteId, bytes6[] calldata path) external auth {
+        // This is hideous, but's the only way to check if a path is already set
+        try compositeOracle.paths(baseId, quoteId, 0) {
+            revert("Path already set");
+        } catch {
+            compositeOracle.setPath(baseId, quoteId, path);
+        }
+    }
+
+    /// ----------------- Ladle Governance -----------------
+
+    /// @notice Propagate a pool to the Ladle from the Yield Ladle
+    function addPool(bytes6 seriesId) external auth returns (IPool pool_) {
+        pool_ = IPool(yieldLadle.pools(seriesId));
+        require(address(pool_) != address(0), "Pool not known to the Yield Ladle");
+        contangoLadle.addPool(seriesId, pool_);
+    }
+
+    /// @notice Propagate an integration to the Ladle from the Yield Ladle
+    function addIntegration(address integration) external auth {
+        contangoLadle.addIntegration(integration, yieldLadle.integrations(integration));
+    }
+
+    /// @notice Propagate a token to the Ladle from the Yield Ladle
+    function addToken(address token) external auth {
```

Comment:
I like `copyX` for when we are propagating configs from the yield ladle to the contango ladle, as you did below.

Reply:
yeah, most of the names were there and I didn't understand the pattern, so didn't change them